pkg/telemetry/metadata.go

```diff
@@ -40,12 +42,21 @@ func promscaleMetadata() Metadata {
 	// leading to type mismatch. Hence, we create a slice to handle both cases.
 	metadata["os_sys_name"] = toString(uname.Sysname[:])
 	metadata["os_node_name"] = toString(uname.Nodename[:])
-	metadata["os_release"] = toString(uname.Release[:])
-	metadata["os_version"] = toString(uname.Version[:])
 	metadata["os_machine"] = toString(uname.Machine[:])
+
+	turnOffPackageLogging()
+	distro := linux.DiscoverDistro()
```

Comment:
You don't need an external dependency for this. You can simply read `/etc/os-release` for distribution name, version, description, and other metadata on all of the major Linux distros (and probably all the minor ones too). This includes Debian/Ubuntu, CentOS/Fedora/RHEL, and Alpine for sure, possibly others.

The file contains newline-separated environment variable definitions, e.g.:

```text
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.14.3
```

On all of the OSs I've tested (all of those I mentioned above) at the very least NAME, ID, and VERSION_ID are set, and in some cases there are additional vars such as `VERSION_CODENAME` (which contains distro names like Debian/Ubuntu use, e.g. `bullseye`).

Since we only care about the distro name and version, `/etc/os-release` covers this well.

Reply:
agreed, but this was a suggestion from Ante, and I do like it since it keeps the code clean and minimal. Plus, the library does the same under the hood, so I don't see anything wrong in using it.

Reply:
@bitwalker It seems the library rules are more exact/specific https://github.com/dekobon/distro-detect/blob/master/linux/distrochecks.go but would be curious if you disagree.